packages/teleterm/src/mainProcess/mainProcess.ts

```diff
@@ -203,6 +203,52 @@ export default class MainProcess {
       this.windowsManager.forceFocusWindow();
     });
 
+    // Used in the `tsh install` command on macOS to make the bundled tsh available in PATH.
+    // Returns true if tsh got successfully installed, false if the user closed the osascript
+    // prompt. Throws an error when osascript fails.
+    ipcMain.handle('main-process-symlink-tsh', async () => {
+      const source = this.settings.tshd.binaryPath;
+      const target = '/usr/local/bin/tsh';
+      const prompt =
+        'Teleport Connect wants to create a symlink for tsh in /usr/local/bin.';
+      const command = `osascript -e "do shell script \\"mkdir -p /usr/local/bin && ln -sf '${source}' '${target}'\\" with prompt \\"${prompt}\\" with administrator privileges"`;
+
+      try {
+        await promisify(exec)(command);
+        this.logger.info(`Created the symlink to ${source} under ${target}`);
+        return true;
+      } catch (error) {
+        // Ignore the error if the user canceled the prompt.
+        // https://developer.apple.com/library/archive/documentation/AppleScript/Conceptual/AppleScriptLangGuide/reference/ASLR_error_codes.html#//apple_ref/doc/uid/TP40000983-CH220-SW2
+        if (error instanceof Error && error.message.includes('-128')) {
+          return false;
+        }
+        this.logger.error(error);
+        throw error;
+      }
+    });
+
+    ipcMain.handle('main-process-remove-tsh-symlink', async () => {
+      const target = '/usr/local/bin/tsh';
+      const prompt =
+        'Teleport Connect wants to remove a symlink for tsh from /usr/local/bin.';
```

Comment:
I wonder if this language could be changed at all. I know this is exactly what the script is doing, and I know our audience is technical enough to understand thing, but it just seems so... destructive? Idk what the word is. I'm ok with not changing it, but I wonder if there is simpler terms to describe what is happening. Maybe I'm overreacting

Reply:
> I'm ok with not changing it, but I wonder if there is simpler terms to describe what is happening.

We could say "Connect wants to uninstall tsh from PATH" but personally I'd prefer the app to tell me exactly what it wants to do when it asks me for a password.

Another option would be "Connect wants to modify the contents of /usr/local/bin" which doesn't use the notion of removing/uninstalling but again fails at specifying what exactly is going to happen and potentially sounds even more dangerous.

I'd probably keep this, the way I look at it is we ask the user for permission to do something so we better explain that as succinctly as possible.